code generators/de.fhdo.lemma.model_processing.code_generation.container_base/src/main/java/de/fhdo/lemma/model_processing/code_generation/container_base/intermediate_model_validation/BuildScriptIntermediateOperationModelValidator.xtend

```diff
@@ -0,0 +1,40 @@
+package de.fhdo.lemma.model_processing.code_generation.container_base.intermediate_model_validation
+
+import de.fhdo.lemma.model_processing.annotations.IntermediateModelValidator
+import de.fhdo.lemma.operation.intermediate.IntermediatePackage
+import de.fhdo.lemma.model_processing.builtin_phases.intermediate_model_validation.AbstractIntermediateDeclarativeValidator
+import de.fhdo.lemma.operation.intermediate.IntermediateContainer
+import org.eclipse.xtext.validation.Check
+
+/**
+ * The build script model validator is responsible for checking the compatibility of the
+ * intermediate operation model with the build script code generation module.
+ *
+ * @author Philip Wizenty(philip.wizenty@fh-dortmund.de)
+ */
+@IntermediateModelValidator
+class BuildScriptIntermediateOperationModelValidator
+    extends AbstractIntermediateDeclarativeValidator {
+
+    /*
+     * Get the namespace for the intermediate operation model
+     */
+    override getLanguageNamespace() {
+        return IntermediatePackage.eNS_URI
+    }
+
+    /*
+     * Check if the deployed microservice uses a maven build technology as the build script
+     * code generation module only supports this technology.
+     */
+    @Check
+    def checkMavenBuildTechnologyForContainer(IntermediateContainer container) {
+        val aspect = container.aspects.findFirst[aspect | aspect.name == "MavenTechnology"]
+        if (aspect.propertyValues.get(0).value == false) {
+            warning("The build script gode generation module only support maven as a build "
+                + "technology",
+                IntermediatePackage.Literals.INTERMEDIATE_CONTAINER__OPERATION_MODEL
```

Comment:
Hmm, I suspect this literal to yield a warning for the _complete_ operation model when actually only the container is concerned. Could you check this?